ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -136,20 +138,25 @@
   private static final String VALID_WINDOW_BOUNDS_TYPES_STRING =
       GrammaticalJoiner.and().join(VALID_WINDOW_BOUNDS_TYPES);
 
-  private final KsqlExecutionContext executionContext;
+  private final KsqlEngine ksqlEngine;
   private final RoutingFilterFactory routingFilterFactory;
   private final RateLimiter rateLimiter;
+  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
 
   public PullQueryExecutor(
-      final KsqlExecutionContext executionContext,
+      final KsqlEngine ksqlEngine,
```

Comment:
Probably better to keep this as `KsqlExecutionContext` and pass in the service Id... less coupling.